### PR TITLE
[Core] Add AEAD_AES_128_GCM and AEAD_AES_256_GCM ciphers support.

### LINKED ISSUE #691: Support for AEAD_256_GCM
I have a Cisco IP8861 phone that seems to implement `AEAD_256_GCM` fro SRTP crypto but not `AEAD_256_GCM_8 `which is what FS supports. When FS passes `AEAD_256_GCM_8` crypto fails to be negotiated. I believe these two crypto schemes are different given that in vars.xml the following is stated: 

```
 AEAD_AES_256_GCM_8
      ____________________________________________________________________________
      This algorithm is identical to AEAD_AES_256_GCM (see Section 5.2 of
      [RFC5116]), except that the tag length, t, is 8, and an
      authentication tag with a length of 8 octets (64 bits) is used.
      An AEAD_AES_256_GCM_8 ciphertext is exactly 8 octets longer than its
      corresponding plaintext.
```

Can support `AEAD_256_GCM` be added ? I'm thinking it shouldn't be too difficult given `AEAD_AES_256_GCM_8` is already supported?

### PR DESCRIPTION
**Pull request by zandeez**
Resolves #691 

Support for this cipher already exists in libsrtp, so simply added matching declrations.
I am not 100% sure on the additions to switch_rtp.h.

Basic testing performed with my Cisco 8841.

### CODE REVIEW

**Comment by zandeez** [comment]
Resolves #691

File: src/switch_core_media.c [hunk 1]
```diff
@@ -278,7 +278,9 @@ struct switch_media_handle_s {
 };
 
 switch_srtp_crypto_suite_t SUITES[CRYPTO_INVALID] = {
+        { "AEAD_AES_256_GCM ", "", AEAD_AES_256_GCM, 44, 12},
```
Comment (andywolk) [inline src/switch_core_media.c:281]: Why does `"AEAD_AES_256_GCM "` have that additional space at the end?

**Comment by zandeez** [comment]
The code that checks what ciphers are in the SDP  is just a simple substring check, so without the space it'd also match AEAD_AES_256_GCM_8 even though they're two distinct ciphers.

Long term it'd be ideal to change the way the SDP is parsed for ciphers so the space can be omitted.

**Comment by andywolk** [comment]
@zandeez Can you please point me to that code that checks ciphers?

**Comment by zandeez** [comment]
No problem, switch_core_media.c, lines 1815 and 1819.

**Comment by andywolk** [comment]
Well not ideal but how about this (swap it)
```
	{ "AEAD_AES_256_GCM_8", "", AEAD_AES_256_GCM_8, 44, 12},
        { "AEAD_AES_256_GCM", "", AEAD_AES_256_GCM, 44, 12},
```
instead of
```
        { "AEAD_AES_256_GCM ", "", AEAD_AES_256_GCM, 44, 12},
	{ "AEAD_AES_256_GCM_8", "", AEAD_AES_256_GCM_8, 44, 12},
```

**Comment by zandeez** [comment]
I will give that a go and report back

**Comment by zandeez** [comment]
With them that way around, 

2020-06-22 11:45:14.924465 [DEBUG] switch_core_media.c:1815 looking for crypto suite [AEAD_AES_256_GCM_8]alias=[] in [3 AEAD_AES_256_GCM inline:cVcOvHPpMUZZr2ruqzKr+4E3I0ujCNxB6ZlbjlvJ5EWRED9v04IDI3nl9r0=]
2020-06-22 11:45:14.924465 [DEBUG] switch_core_media.c:1815 looking for crypto suite [AEAD_AES_256_GCM]alias=[] in [3 AEAD_AES_256_GCM inline:cVcOvHPpMUZZr2ruqzKr+4E3I0ujCNxB6ZlbjlvJ5EWRED9v04IDI3nl9r0=]
2020-06-22 11:45:14.924465 [DEBUG] switch_core_media.c:1824 Found suite AEAD_AES_256_GCM
2020-06-22 11:45:14.924465 [DEBUG] switch_core_media.c:1894 Set Remote Key [3 AEAD_AES_256_GCM inline:cVcOvHPpMUZZr2ruqzKr+4E3I0ujCNxB6ZlbjlvJ5EWRED9v04IDI3nl9r0=]
2020-06-22 11:45:14.924465 [DEBUG] switch_core_media.c:1217 Set Local audio crypto Key [3 AEAD_AES_256_GCM_8 inline:Y+TUI72KLeuUVZ3xDDjdXaR7sJOM9UqtP70lFxjAS8CX3YAUWo+kEntdG7Q=]

The FreeSWITCH end sends AEAD_AES_256_GCM_8 even though the client requests AEAD_AES_256_GCM.

**Comment by andywolk** [comment]
Did you swap in `switch_types.h` as well?

**Comment by zandeez** [comment]
I have now and that seems to have sorted it, I'll sort out the commit shortly.

**Pull request merged**